internal/helm/repository/oci_chart_repository.go

```diff
@@ -121,7 +121,7 @@ func NewOCIChartRepository(repositoryURL string, chartRepoOpts ...OCIChartReposi
 func (r *OCIChartRepository) Get(name, ver string) (*repo.ChartVersion, error) {
 	// Find chart versions matching the given name.
 	// Either in an index file or from a registry.
-	cvs, err := r.getTags(fmt.Sprintf("%s/%s", r.URL.String(), name))
+	cvs, err := r.getTags(fmt.Sprintf("%s/%s", strings.TrimSuffix(r.URL.String(), "/"), name))
```

Comment:
Since this is constructing a URL, how about using `path.Join()` to be more safe, https://pkg.go.dev/path#Join .

Reply:
I also think that this should be done earlier. Maybe just before or in `NewOCIChartRepository` call.

Reply:
I think we can do both. The above suggestion would just make the code safer. Separately, we can normalize the URL before reaching here.

Reply:
Need to also update how URL is constructed in https://github.com/fluxcd/source-controller/blob/7c07b7a03fdafba280e11c90e36cc4e53f431da6/internal/helm/repository/oci_chart_repository.go#L139 for a proper fix using `path.Join()`.

Reply:
I have updated the pull request to do both.
I also changed the methods for `OCIChartRepository` to take a value receiver instead of a pointer since none of the methods modify it.